src/lib/InstantSearch.js

```diff
@@ -296,87 +236,70 @@ See: https://www.algolia.com/doc/guides/building-search-ui/widgets/create-your-o
       );
       this._createURL = routingManager.createURL.bind(routingManager);
       this._createAbsoluteURL = this._createURL;
-      this.widgets.push(routingManager);
+      // We don't use `addWidgets` because we have to ensure that `RoutingManager`
+      // is the last widget added. Otherwise we have an issue with the `routing`.
+      // https://github.com/algolia/instantsearch.js/pull/3149
+      this.mainIndex.getWidgets().push(routingManager);
```

Comment:
Not sure I understood the statement. Indeed `.widgets` is not available anymore since it's now scoped under the "main" index. The schema provided is incorrect. The `router` has to be scoped under `index` otherwise it does not have access to the rest of the widgets.

Note that for compatibly purpose we can still expose it. I didn't do it by default because (I think) the usage of `.widget` isn't that common. We can mirror the value of the main index if required (like we do for the [`helper`](https://github.com/algolia/instantsearch.js/pull/3893#discussion_r297712527)).

Reply:
If the router needs to be scoped under the main index, how does it work for any lower indices then? This is why I ask how come it can't be at the root instead

Reply:
Yes that's right it doesn't work. At the moment the multi-index does not work anyway (we don't merge the parameters). The changes inside this PR mirror the current implementation with the usage of a main index. The routing will be updated to support multiple levels.